data/uhc_pack/functions/running/diamond_chestplate/on_obtain.mcfunction

```diff
@@ -0,0 +1,5 @@
+advancement revoke @s only uhc_pack:running/obtain_diamond_chestplate
+
+# Mark the player who received the chestplate on a scoreboard.
+# This means cleanup can be handled the tick after, which eliminates desync issues
+execute if score UHCDiamondChestplate uhcEnabled matches 0 run scoreboard players set @s uhcDiaChest 1
```

Comment:
Instead of using a scoreboard, it would be better to use `schedule function uhc_pack:running/diamond_chestplate/cleanup 1t`

EDIT: Actually scratch that. You will loose the context player if you do this (it would be great if the schedule command could remember the entity)